### PR TITLE
Update the upgrade guide for Query classes

### PR DESCRIPTION
**Pull request by markstory**
Add the upgrade notes for the new query subclasses to help with forwards compatibilty. I'll update all the code examples later.

### CODE REVIEW

File: en/appendices/4-5-migration-guide.rst [hunk 1]
```diff
@@ -32,6 +32,35 @@ features::
...
+The new classes are:
+
+- ``Cake\ORM\Query\DeleteQuery`` Used for building ``delete`` queries.
+- ``Cake\ORM\Query\InsertQuery`` Used for building ``insert`` queries.
+- ``Cake\ORM\Query\SelectQuery`` Used for building ``select`` queries.
+- ``Cake\ORM\Query\UpdateQuery`` Used for building ``update`` queries.
+
+Each of these classes no longer offer methods that don't make sense for that
+query type. For example, ``DeleteQuery`` has no ``select()`` clause. In 4.5,
+calling ``select()`` on a query intended to be used as a delete.
```
Comment (othercorey) [inline en/appendices/4-5-migration-guide.rst]: This is missing part of the sentence.

File: en/appendices/4-5-migration-guide.rst [hunk 2]
```diff
@@ -32,6 +32,35 @@ features::
...
+
+To help you adopt the new query interfaces and define the kind of query you want
```
Comment (othercorey) [inline en/appendices/4-5-migration-guide.rst:54]:
```suggestion
To help you adopt the new query API and define the kind of query you want
```

File: en/appendices/4-5-migration-guide.rst [hunk 3]
```diff
@@ -32,6 +32,35 @@ features::
...
+earlier. To encourage planning ahead, ``Table::query()`` is deprecated.
+Replacing it are new methods on ``Table``. The ``deleteQuery()``,
```
Comment (othercorey) [inline en/appendices/4-5-migration-guide.rst]:
```suggestion
The new ``deleteQuery()``,
```

**Review by Arhell — APPROVED** [review]

File: en/appendices/4-5-migration-guide.rst (hunk 1 above)
Comment (markstory) [reply en/appendices/4-5-migration-guide.rst]: There are a few nonsensical things here. I will try to do better in the future.

**Comment by markstory** [comment]
Going to hold off on merging this until I get the `Connection` methods added.

**Pull request merged**